src/proxy.rs

```diff
@@ -100,20 +104,20 @@ impl Proxy {
     /// # fn run() -> Result<(), Box<::std::error::Error>> {
     /// let target = reqwest::Url::parse("https://my.prox")?;
     /// let client = reqwest::Client::builder()
-    ///     .proxy(reqwest::Proxy::custom(move |url| {
-    ///         if url.host_str() == Some("hyper.rs") {
-    ///             Some(target.clone())
-    ///         } else {
-    ///             None
-    ///         }
-    ///     }))
+    ///     .proxy(reqwest::Proxy::custom(|url| url.host() == Some("hyper.rs"),
+    ///                                   "http://proxy.custom")?)
     ///     .build()?;
     /// # Ok(())
     /// # }
     /// # fn main() {}
-    pub fn custom<F>(fun: F) -> Proxy
-    where F: Fn(&Url) -> Option<Url> + Send + Sync + 'static {
-        Proxy::new(Intercept::Custom(Custom(Arc::new(fun))))
+    pub fn custom<F, U: IntoUrl>(fun: F, url: U) -> ::Result<Proxy>
+    where F: Fn(&Uri) -> bool + 'static + Send + Sync {
+        Proxy::new(fun, url)
+    }
+
+    /// Set proxy authorization
+    pub fn set_authorization<S: Scheme + Any>(&mut self, scheme: S) {
```

Comment:
Just curious, what is the `Any` bound for? Is it possible to remove it?

Reply:
This is required by hyper:
https://docs.rs/hyper/0.11.10/hyper/header/struct.Authorization.html
https://docs.rs/hyper/0.11.10/hyper/header/struct.ProxyAuthorization.html